src/reducers/versions.spec.tsx

```diff
@@ -352,6 +352,30 @@ describe(__filename, () => {
       expect(_log.debug).toHaveBeenCalled();
       expect(versionsState).toEqual(previousState);
     });
+
+    it('logs an error message when headVersion is missing on loadDiff()', () => {
+      const addonId = 1;
+      const baseVersionId = 2;
+      const version = fakeVersionWithDiff;
+      const headVersionId = version.id;
+      const _log = createFakeLogger();
+
+      const versionsState = reducer(
+        undefined,
+        actions.loadDiff({
+          addonId,
+          baseVersionId,
+          headVersionId,
+          version,
+        }),
+        // TS looks confused about the third optional argument.
+        // @ts-ignore
+        { _log },
```

Comment:
It makes sense that TS won't allow this because Redux has a specific reducer signature and this doesn't match it. We can make a pretty good guess that the reducer will still work in the wild with a `_log` option *but* I think it would be fine to honor the type signature and just change the design. We could do it like this:

```tsx
const createReducer = ({ _log = log } = {}) => {
  const reducer: Reducer<VersionsState, ActionType<typeof actions>> = (
    state = initialState,
    action,
  ): VersionsState => {
    switch (action.type) {
      // ...
    }
  };

  return reducer;
};

const reducer = createReducer();
export default reducer;
```

This might be super annoying -- just throwing it out as an idea (not a change request).